options/posix/generic/arpa-inet-stubs.cpp

```diff
@@ -44,18 +69,64 @@ char *inet_ntoa(struct in_addr) {
 	__ensure(!"Not implemented");
 	__builtin_unreachable();
 }
-int inet_aton(const char *, struct in_addr *) {
-	__ensure(!"Not implemented");
-	__builtin_unreachable();
+int inet_aton(const char *string, struct in_addr *dest) {
+	char *dest_array = reinterpret_cast<char*>(dest);
+	int array[4];
+	int i = 0;
+	char *save;
+
+	for (; i < 4; i++) {
+		array[i] = strtoul(string, &save, 0);
+		if (save == string || (*save && *save != '.') || isdigit(*save))
+			return 0;
+		if (!*save) break;
+		string = save + 1;
+	}
+
+	switch (i) {
+		case 0:
+			array[1] = array[0] & 0xffffff;
+			array[0] >>= 24;
+		case 1:
+			array[2] = array[1] & 0xffff;
+			array[1] >>= 16;
+		case 2:
+			array[3] = array[2] & 0xff;
+			array[2] >>= 8;
+	}
+
+	for (i = 0; i < 4; i++) {
+		if (array[i] > 255)
+			return 0;
+		dest_array[i] = array[i];
```

Comment:
Instead of using `reinterpret_cast` and type punning, we can build the `struct in_addr` using bit shifts.